cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1399,6 +1411,52 @@ if(ARROW_JEMALLOC)
   add_dependencies(toolchain jemalloc_ep)
 endif()
 
+# ----------------------------------------------------------------------
+# mimalloc - Cross-platform high-performance allocator, from Microsoft
+
+if(ARROW_MIMALLOC)
+  message(STATUS "Building (vendored) mimalloc from source")
+  # We only use a vendored mimalloc as we want to control its build options.
+
+  # XXX Careful: mimalloc library naming varies depend on build type capitalization:
+  # https://github.com/microsoft/mimalloc/issues/144
+  string(TOLOWER ${CMAKE_BUILD_TYPE} _lowercase_build_type)
```

Comment:
Could you define `LOWERCASE_BUILD_TYPE` https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L31 and use it?